x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/security/authc/SubjectTests.java

```diff
@@ -167,9 +166,9 @@ public void testGetRoleReferencesForRemoteAccess() {
         );
         authMetadata.put(AuthenticationField.API_KEY_LIMITED_ROLE_DESCRIPTORS_KEY, limitedByRoleBytes);
 
-        final RemoteAccessAuthentication remoteAccessAuthentication = randomBoolean()
-            ? AuthenticationTestHelper.randomRemoteAccessAuthentication(RoleDescriptorsIntersection.EMPTY)
-            : AuthenticationTestHelper.randomRemoteAccessAuthentication();
+        final RemoteAccessAuthentication remoteAccessAuthentication = AuthenticationTestHelper.randomRemoteAccessAuthentication(
+            randomIntBetween(0, 1)
```

Comment:
Since we've now enabled API key support, would also optional test with an API key for this test (and consequently 2 role descriptors here instead of one, when dealing with API keys).

Reply:
Makes sense. I will change this.